api/configuration/v1alpha1/kong_certificate.go

```diff
@@ -0,0 +1,141 @@
+package v1alpha1
+
+import (
+	konnectv1alpha1 "github.com/kong/kubernetes-configuration/api/konnect/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// KongCertificate is the schema for Certificate API which defines a Kong Certificate.
+//
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Programmed",description="The Resource is Programmed on Konnect",type=string,JSONPath=`.status.conditions[?(@.type=='Programmed')].status`
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.spec.controlPlaneRef) || has(self.spec.controlPlaneRef)", message="controlPlaneRef is required once set"
+// +kubebuilder:validation:XValidation:rule="(!self.status.conditions.exists(c, c.type == 'Programmed' && c.status == 'True')) ? true : oldSelf.spec.controlPlaneRef == self.spec.controlPlaneRef", message="spec.controlPlaneRef is immutable when an entity is already Programmed"
+type KongCertificate struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec KongCertificateSpec `json:"spec"`
+
+	// +kubebuilder:default={conditions: {{type: "Programmed", status: "Unknown", reason:"Pending", message:"Waiting for controller", lastTransitionTime: "1970-01-01T00:00:00Z"}}}
+	Status KongCertificateStatus `json:"status,omitempty"`
+}
+
+// GetKonnectStatus returns the Konnect status contained in the KongRoute status.
```

Comment:
All these Konnect related funcs should be generated with https://github.com/Kong/kubernetes-configuration/tree/main/scripts/konnect-funcs

as stated in the README: https://github.com/Kong/kubernetes-configuration/blob/d1106e28743b5c87d4adb5d36725251c4f2581ff/README.md#L52-L53

Reply:
Thank you!